tests/model_composite/tests.py

```diff
@@ -0,0 +1,95 @@
+from django.test import TestCase
+
+from .models import Employee, Employee2, User
+
+
+class CompositePKTests(TestCase):
+    def test_custom_pk_create(self):
+        """
+        New objects can be created both with pk and the custom name
+        """
+
+        self.assertEqual(Employee(pk=("root", 1235)).pk, ("root", 1235))
+        self.assertEqual(Employee2(pk=("root", 1235)).pk, ("root", 1235))
+
+        self.assertEqual(Employee(composite_pk=("root", 1235)).pk, ("root", 1235))
+        self.assertEqual(Employee2(composite_pk=("root", 1235)).pk, ("root", 1235))
+
+        self.assertEqual(Employee().pk, ("", None))
+        self.assertEqual(Employee2().pk, ("", None))
+
+        self.assertEqual(Employee(composite_pk=("root", 1235)).pk, ("root", 1235))
+        self.assertEqual(Employee2(composite_pk=("root", 1235)).pk, ("root", 1235))
+
+        self.assertRaises(TypeError, Employee, composite_pk=("root", 1235), branch="")
+        self.assertRaises(TypeError, Employee2, composite_pk=("root", 1235), branch="")
```

Comment:
I'd use the context manager version of `assertRaises`, I think it's a lot more readable:

```suggestion
        with self.assertRaises(TypeError):
            Employee(composite_pk=("root", 1235), branch="")
        with self.assertRaises(TypeError):
            Employee2(composite_pk=("root", 1235), branch="")
```